platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r940/ResilientKotlinDslScriptsModelBuilderCrossVersionSpec.groovy

```diff
@@ -849,18 +847,34 @@ class ResilientKotlinDslScriptsModelBuilderCrossVersionSpec extends ToolingApiSp
     }
 
     static class SetStartParameterAction implements BuildAction<String>, Serializable {
+
+        private final boolean resilient;
+
+        SetStartParameterAction(boolean resilient) {
+            this.resilient = resilient
+        }
+
         @Override
         String execute(BuildController controller) {
-            def gradleBuild = controller.getModel(GradleBuild)
-            def result = controller.fetch(gradleBuild.rootProject, StartParametersModel)
-            return result.failures.isEmpty() ? "successful" : "unsuccessful"
+            if (resilient) {
+                def gradleBuild = controller.fetch(GradleBuild).model
+                if (gradleBuild) {
+                    def result = controller.fetch(gradleBuild.rootProject, StartParametersModel)
+                    return result.failures.isEmpty() ? "successful" : "unsuccessful"
+                }
+                return "unsuccessful"
+            } else {
+                def gradleBuild = controller.getModel(GradleBuild)
+                def result = controller.getModel(gradleBuild.rootProject, StartParametersModel)
+                return result
+            }
         }
     }
 
     static class KotlinModelOnNullTargetAction implements BuildAction<KotlinModel>, Serializable {
         @Override
         KotlinModel execute(BuildController controller) {
-            GradleBuild build = controller.getModel(GradleBuild.class)
```

Comment:
💅 I would just assert that build is not null, so we get maybe a better error message if that happens (if would fail later in `queryResilientKotlinDslScriptsModel` with NPE anyway though)

Reply:
Done.